frontend/server/k8s-helper.ts

```diff
@@ -127,3 +132,79 @@ export function getPodLogs(podName: string): Promise<string> {
       (error: any) => { throw new Error(JSON.stringify(error.body)); }
     );
 }
+
+/**
+* Create Tensorboard pod via CRD with the given logdir if there is no existing
+* Tensorboard pod.
+*/
+export async function newTensorboardInstance(logdir: string): Promise<void> {
+  if (!k8sV1CustomObjectClient) {
+    throw new Error('Cannot access kubernetes Custom Object API');
+  }
+  const currentPod = await getTensorboardInstance(logdir);
+  if (currentPod) {
+    return;
+  }
+
+  // TODO: take the configuration below to a separate file
+  // Name of the viewer resource is based on logDir.
+  const group = 'kubeflow.org';
+  const version = 'v1beta1';
+  const plural = 'viewers';
+  const body = {
+    apiVersion: group + '/' + version,
+    kind: 'Viewer',
+    metadata: {
+      name: 'viewer-' + crypto.SHA1(logdir),
```

Comment:
can you extract the 'viewer-' + crypto.SHA1(logdir) into a function? and reuse the function below

Reply:
done